### PR TITLE
tea-index: Adding a rough draft based on discussions in Koala wg meeting

### PR DESCRIPTION
**Pull request by oej**
A first draft documenting some of the decisions made in the meeting June 19th, 2024 - including some open issues.

### CODE REVIEW

File: tea-index/README.md [hunk 1]
```diff
@@ -0,0 +1,69 @@
+# The TEA index API
```
Comment (dasarpjonam) [inline tea-index/README.md:1]: Is TEA Index similar to [Certificate Transparency Log aka ctlog ](https://certificate.transparency.dev/)? do we assume that the exchange index stores products from different producers or 1 exchange index stores 1 producer related products?

File: tea-index/README.md [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+The format of the version may follow many syntaxes, so maybe
+the API needs to be able to provide some sort of format
+for the version string.
+
+An automated system may want to provide the user with a GUI,
+listing versions and being able to scroll to the next page
+until the user selects a version.
+
+### Content of the index object
+
```
Comment (dasarpjonam) [inline tea-index/README.md:43]: should the response be a collection of CDX objects? I assumed UUID, Identifier, metadata are all CDX attributes.

File: tea-index/README.md (hunk 1 above)
Comment (oej) [reply tea-index/README.md:1]: No. I can explain further in Slack. The index is just for ONE single product from ONE single vendor. It's an index over all versions of a specific software.

File: tea-index/README.md (hunk 2 above)
Comment (oej) [reply tea-index/README.md:43]: The TEA is not locked to CDX. You can get a collection of artefacts - like In-Toto attestations, SPDX files, CycloneDX BOMs and other docs - like the EU certificate of conformance.

**Pull request merged**